### PR TITLE
Adds fail state to gear tweak unit test.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
Adds a return value to gear tweaks and checks it in the unit test.

## Why and what will this PR improve
Runtimes two procs deep were not being reported as failures.

## Authorship
@MarinaGryphon identified the issue, I wrote the code.

## Changelog
Nothing player facing.

### CODE REVIEW

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -24,6 +24,8 @@
 /mob/new_player/Initialize()
 	. = ..()
 	verbs += /mob/proc/toggle_antag_pool
+	loc = null
```
Comment (afterthought2) [inline code/modules/mob/new_player/new_player.dm]: I know I left it alone previously, but possibly this should be `forceMove(null)` or even `ASSERT(loc == null)`.

**Review by afterthought2 — DISMISSED** [review]

File: code/modules/mob/new_player/new_player.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/mob/new_player/new_player.dm]: I considered that but wasn't sure if it was deliberately skipping Entered() etc being called.

Comment (afterthought2) [reply code/modules/mob/new_player/new_player.dm]: I don't know; they should never be spawned outside of the client/New anyway, but I guess you can bus them in (?).

Comment (MistakeNot4892) [reply code/modules/mob/new_player/new_player.dm]: Removed these changes from this PR in any case.

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**